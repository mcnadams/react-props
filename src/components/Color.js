import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, rgb }) {
  return (
    <dl>

      {name && <><dt>Name:</dt>
      <dd>{name}</dd></>}

      <dt>Hex:</dt>
      <dd>{hex}</dd>

      <dt>RGB:</dt>
      <dd>Red: {rgb.red}</dd>
      <dd>Green: {rgb.green}</dd>
      <dd>Blue: {rgb.blue}</dd>
        
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.object.isRequired
};

export default Color;
