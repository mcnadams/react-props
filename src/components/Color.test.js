import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color name='fuschia' hex='#CA2C92' rgb={{
      red: 202,
      green: 44,
      blue: 126
    }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
