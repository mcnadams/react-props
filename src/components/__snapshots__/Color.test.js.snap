// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl>
  <dt>
    Name:
  </dt>
  <dd>
    fuschia
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    #CA2C92
  </dd>
  <dt>
    RGB:
  </dt>
  <dd>
    Red: 
    202
  </dd>
  <dd>
    Green: 
    44
  </dd>
  <dd>
    Blue: 
    126
  </dd>
</dl>
`;
